### PR TITLE
Replace some crypto, usage of rand

### PR DESCRIPTION
**Pull request by ThomasdenH**
This PR replaces some crypto (SHA2 and MD5) by external crates and uses the rand crate for some random number generation. It also rustifies some things in `dpx_pdfencrypt.rs`.

### CODE REVIEW

File: dpx/src/dpx_pdfencrypt.rs [hunk 1]
```diff
@@ -128,79 +126,54 @@ static mut verbose: u8 = 0_u8;
...
+        bd_time.tm_year + 1900,
+        bd_time.tm_mon + 1,
+        bd_time.tm_mday,
+        bd_time.tm_hour,
+        bd_time.tm_min,
+        bd_time.tm_sec,
+    ));
+    md5.input(&format!(
+        "{}-{}, Copyright 2002-2015 by Jin-Hwan Cho, Matthias Franz, and Shunsaku Hirata",
+        // TODO: Are these variables?
```
Comment (cormacrelf) [inline dpx/src/dpx_pdfencrypt.rs:166]: I would guess compile time macros that got flattened

File: dpx/src/dpx_pdfencrypt.rs [hunk 2]
```diff
@@ -128,79 +126,54 @@ static mut verbose: u8 = 0_u8;
...
     let mut current_time: time_t = 0;
     let p = &mut sec_data;
     current_time = get_unique_time_if_given();
-    if current_time == -1i32 as time_t {
-        current_time = time(0 as *mut time_t)
+    if current_time == -1 {
+        current_time = time(std::ptr::null_mut())
     }
+    // TODO: libc rand is not used in this module
     srand(current_time as u32);
```
Comment (cormacrelf) [inline dpx/src/dpx_pdfencrypt.rs:138]: rand() is used in the AES routine for the time being.

**Review by cormacrelf — COMMENTED** [review]
A few unused `buf_get_be32` etc functions left hanging around in dpxcrypt but they'll hopefully be gone soon too, good work

**Comment by crlf0710** [comment]
Looks great, let's merge this, thank you!

**Comment by cormacrelf** [comment]
Linking to #71 as partially done

**Pull request merged**